### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by abughalib**

### CODE REVIEW

File: handlers/interactions/buttonInteractions.ts [hunk 1]
```diff
@@ -41,23 +42,20 @@ async function interactionButtonHandler(interaction: ButtonInteraction) {
...
 
     let joined_user = "";
 
-    let originalFields: APIEmbedField[] =
-      interaction.message.embeds[0].data.fields;
-
     let spots: number = 0;
 
-    originalFields.forEach((field) => {
+    for (const field of interaction.message.embeds[0].data.fields) {
```
Comment (AshAman999) [inline handlers/interactions/buttonInteractions.ts]: u can use perhaps a simple for loop

File: utils/helpers.ts [hunk 2]
```diff
@@ -24,4 +27,216 @@ function formatTime(duration: number): string {
   return `About ${duration} Hours`;
 }
```
Comment (AshAman999) [inline utils/helpers.ts:33]: You could perhaps use comments to specify that it's for a certain game

File: handlers/interactions/buttonInteractions.ts (hunk 1 above)
Comment (abughalib) [reply handlers/interactions/buttonInteractions.ts]: Done

File: handlers/interactions/buttonInteractions.ts [hunk 3]
```diff
@@ -338,9 +339,9 @@ async function interactionButtonHandler(interaction: ButtonInteraction) {
 
     team_players = removeEntry(team_players, interaction.user.toString());
 
-    for (const field of new_fields) {
-      if (field.name === "Players Joined") {
-        field.value = team_players.join("\n") || "";
+    for (let i = 0; i < new_fields.length; i += 1) {
```
Comment (AshAman999) [inline handlers/interactions/buttonInteractions.ts:518]: Can u add a comment why, this logic here , seems confusing to me, what actually it does

File: handlers/interactions/buttonInteractions.ts [hunk 4]
```diff
@@ -43,17 +43,18 @@ async function interactionButtonHandler(interaction: ButtonInteraction) {
...
 
-    for (const field of interaction.message.embeds[0].data.fields) {
-      if (field.name === "Number of Space in Wing/Team Available") {
-        spots = parseInt(field.value);
-      } else if (field.name === "Players Joined") {
-        joined_user = field.value;
+    for (let i = 0; i < fields.length; i += 1) {
+      if (fields[i].name === "Number of Space in Wing/Team Available") {
+        spots = parseInt(fields[i].value);
+      } else if (fields[i].name === "Players Joined") {
```
Comment (AshAman999) [inline handlers/interactions/buttonInteractions.ts:82]: same , add comment

**Review by AshAman999 — CHANGES_REQUESTED** [review]
Please add comments, it looks really confusing

File: utils/helpers.ts (hunk 2 above)
Comment (abughalib) [reply utils/helpers.ts:33]: Done.

**Review by AshAman999 — APPROVED** [review]
Looks somewhat good now , would still say to put all the stuff related to one game in one directory, if that's possible.

Good for now ✅

**Pull request merged**